### PR TITLE
Update volume's passphrase to null if diskOffering doesn't support encryption

### PR DESCRIPTION
**Pull request by vishesh92**
### Description

This PR updates the volume's passphrase to null if diskOffering doesn't support encryption

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [x] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by vishesh92** [comment]
@blueorangutan package

File: engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestrator.java [hunk 1]
```diff
@@ -1720,6 +1720,9 @@ private Pair<VolumeVO, DataStore> recreateVolume(VolumeVO vol, VirtualMachinePro
             DiskOffering diskOffering = _entityMgr.findById(DiskOffering.class, vol.getDiskOfferingId());
             if (diskOffering.getEncrypt()) {
                 vol = setPassphraseForVolumeEncryption(vol);
+            } else if (vol.getPassphraseId() != null) {
```
Comment (sureshanaparti) [inline engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestrator.java:1723]: @vishesh92 I think, better to not allow offering change for encrypted volumes if disk offering doesn't support encryption.

File: engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestrator.java [hunk 2]
```diff
@@ -1720,6 +1720,9 @@ private Pair<VolumeVO, DataStore> recreateVolume(VolumeVO vol, VirtualMachinePro
...
+                vol.setPassphraseId(null);
+                _volsDao.persist(vol);
```
[the remark this answers is not in the document]
Comment (vishesh92) [reply engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestrator.java]: Updated my changes

**Comment by vishesh92** [comment]
@blueorangutan package

**Comment by vishesh92** [comment]
@blueorangutan package

**Review by sureshanaparti — APPROVED** [review]
code lgtm

**Comment by sureshanaparti** [comment]
@blueorangutan test

**Review by rohityadavcloud — APPROVED** [review]

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Review by borisstoyanov — APPROVED** [review]
LGTM, it now stores NULL for volumes with no encryption

```
MariaDB [cloud]> select disk_offering_id,passphrase_id from volumes;
+------------------+---------------+
| disk_offering_id | passphrase_id |
+------------------+---------------+
|                9 |          NULL |
|               13 |          NULL |
|                1 |          NULL |
|                3 |          NULL |
+------------------+---------------+
4 rows in set (0.00 sec)

MariaDB [cloud]> select name,encrypt from disk_offering where id in (1,3);
+----------------+---------+
| name           | encrypt |
+----------------+---------+
| Small Instance |       0 |
| Small          |       0 |
+----------------+---------+
2 rows in set (0.00 sec)
```

**Pull request merged**